linera-base/src/identifiers.rs

```diff
@@ -493,10 +527,16 @@ impl<Abi: PartialOrd, Parameters, InstantiationArgument> PartialOrd
 {
     fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
         let BytecodeId {
-            message_id,
+            contract_blob_hash: contract_blob_id,
+            service_blob_hash: service_blob_id,
             _phantom,
         } = other;
-        self.message_id.partial_cmp(message_id)
+        let cmp = self.contract_blob_hash.partial_cmp(contract_blob_id);
+        if cmp == Some(std::cmp::Ordering::Equal) {
+            self.service_blob_hash.partial_cmp(service_blob_id)
+        } else {
+            cmp
+        }
```

Comment:
```suggestion
        Some(self.cmp(other))
```

Reply:
I can only access `cmp` here if I add `+ Ord` to the `Abi` generic for this, which means everywhere that uses `BytecodeId`'s `partial_cmp` would also have to implement `Ord`... which seems unusual? Seems like even though this is more code, it at least keeps the implementations separate 🤔
What do you think?

Reply:
We shouldn't even need the `Abi: PartialOrd` bound, neither here nor for the `Ord` impl. If we remove it from both, `Some(self.cmp(other))` should work?

Reply:
We do need the `PartialOrd` bound, because we're implementing `PartialOrd`. If I remove the bound, the code doesn't compile 🤔

Reply:
Oh, my bad. If I actually remove the bounds from all the custom implementations it compiles 😅